platform/api/oci/types/image.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+type CreateImageInput struct {
+	CompartmentID string             `json:"compartmentId"`
+	DisplayName   *string            `json:"displayName,omitempty"`
+	ImageSource   ImageSourceDetails `json:"imageSourceDetails,omitempty"`
+	InstanceID    *string            `json:"instanceId,omitempty"`
+}
+
+type ImageSourceDetails interface{}
```

Comment:
Why is this an `interface{}`?

Reply:
It's used as a base for `ImageSourceViaObjectStorageTuple` & `ImageSourceViaObjectStorageURI` because `CreateImageInput` can return either. Per other discussions in the thread though I'm going to drop `ImageStorageViaObjectStorageTuple` and will drop the Interface as a result.

Reply:
For that case, IMO, you'd want to use `json.RawMessage` and have the type it's in have a clarifying name (e.g. `XMessageWrapper`).. and then have the actual proper types specified out as non-wrappers.